platform/source/mbed_retarget.cpp

```diff
@@ -783,16 +899,35 @@ extern "C" ssize_t read(int fildes, void *buf, size_t length)
     }
 }
 
+#if MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY && DEVICE_SERIAL
```

Comment:
`DEVICE_SERIAL` test shouldn't be here. It should be on the `if MBED_CONF_TARGET_CONSOLE_UART` below.